lib/luadata.c

```diff
@@ -35,6 +38,12 @@ static inline luadata_t *luadata_checkdata(lua_State *L, lua_Integer *offset, lu
 
 #define luadata_checkint(L, offset, T)	luadata_checkdata((L), &(offset), sizeof(T##_t))
 
+static inline void luadata_checkwritable(lua_State *L, luadata_t *data)
+{
+	if (data->opt & LUADATA_OPT_READONLY)
+		luaL_error(L, "read only data");
+}
+
```

Comment:
in this case, of adding "data" to the error message, I would use argcheck instead, so it will print the `data` object as well giving as a better error handling.. 
```suggestion
#define luadata_checkwritable(L, data, ix)	luaL_argcheck((L), (data)->opt & LUADATA_OPT_READONLY), 2, "read only")
```